CONTRIBUTING.md

```diff
@@ -0,0 +1,86 @@
+You may obtain a copy of the License at
+
+   http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+-->
+
+# Contributing Guide for k8s-bigip-ctlr
+If you have found this that means you you want to help us out. Thanks in advance for lending a hand! This guide should get you up and running quickly and make it easy for you to contribute.  
+
+## Issues
+Creating issues is good, creating good issues is even better. Filing meaningful bug reports with lots of information in them helps us figure out what to fix when and how it impacts our users. We like bugs because it means people are using our code, and we like fixing them even more.
+
+Please follow these guidelines for filing issues.
+* Describe the problem
+* Include version of k8s, k8s-bigip-ctlr, and BIG-IP
+* Include detailed information about how to recreate the issue
+* Include relevant and sanitized (no identification) configuration, error messages and logs
+
+## Pull Requests
+The hooks that are automatically run as part of each pull request are:
+* Unit tests are executed
+* Code coverage data is collected
+* Formatting and linting checks are performed
+* Third party attributions are generated and checked
+* Product documentation is built
+* Container is built and pushed to dockerhub
+
+If you are submitting a pull request, you need to make sure that you have done a few things first.
+
+* If an issue doesn't exist, file one
+* Make sure you have tested your code
+* The master branch must be kept release ready at all times. This requires that a single commit should contain the code changes, unit tests, and documentation changes (if any)
+* Use proper formatting for the code
+* Clean up your git history because no one wants to see 75 commits for one issue
+*  Use the commit message format shown below
+
+## Commit Message Format
+The commit message for your final commit should use the following format:
+```
+<One line summarizing what changed>
+
+Problem: Brief description of the problem.
+
+Solution: Detailed description of your solution
+
+Testing (optional if not described in Solution section): Description of tests that were run to exercise the solutions (unit tests, system tests, etc)
+
+Fixes#<issue-number> (Or Fixes: #<issue number> if this commit fixes it)
+
+affects-branches: branch1, branch2
+```
+* The messages should be line-wrapped to 80 characters if possible. Try to keep the one line summary under 80 characters if possible.
+* If a commit fixes many issues, list all of them
+* A line stating what branches the pull request is going to be merged into is required. The note should follow the format "affects-branches: branch1, branch2".
```

Comment:
You might want to put a note here that says this is because we have a robot that can check if bugfixes have been appropriately backported.  This is only needed for bugfixes, and that if you don't know what to put here for a bug, ask in your pull request.